contracts/validator/ValidatorSet.sol

```diff
@@ -50,4 +50,80 @@ contract ValidatorSet {
      *   - less than MAX_FUTURE_END_HEIGHT: for logged out validators.
      */
     mapping(address => uint256) public validatorEndHeight;
+
+
+    /* Internal functions  */
+
+    /**
+     * @notice It is for inserting validators into the validator set.
+     *
+     * @dev Function requires :
+     *          - Validator address must not be 0.
+     *          - Validator address is already used.
+     *
+     * @param _validator Validator address.
+     * @param _beginHeight Begin height for the validator.
+     */
+    function insertValidator(
+        address _validator,
+        uint256 _beginHeight
+    )
+        internal
+    {
+        require(
+            _validator != address(0),
+            "Validator must not be null address."
+        );
+        require(
+            validatorBeginHeight[_validator] == 0,
+            "Validator address is already used."
+        );
+
+        // First validator is being added.
+        if(validators[SENTINEL_VALIDATORS] == address(0)) {
+            validators[SENTINEL_VALIDATORS] = _validator;
+            validators[_validator] = SENTINEL_VALIDATORS;
+        }
+        else {
+            address currentValidator = validators[SENTINEL_VALIDATORS];
+            validators[_validator] = currentValidator;
+            validators[SENTINEL_VALIDATORS] = _validator;
+        }
+
+        validatorBeginHeight[_validator] = _beginHeight;
+        validatorEndHeight[_validator] = MAX_FUTURE_END_HEIGHT;
+    }
+
+    /**
+     * @notice It is for removing validators from the validator set.
+     *
+     * @dev Function requires :
+     *          - Validator address must not be 0.
+     *          - Validator begin height must be less than end height.
+     *          - Validator end height must be equal to MAX_FUTURE_END_HEIGHT.
+     *
+     * @param _validator Validator address.
+     * @param _endHeight End height for the validator.
+     */
+    function removeValidator(
+        address _validator,
+        uint256 _endHeight
+    )
+        internal
+    {
+        require(
```

Comment:
This is not actually needed as there is no validator with 0 address inserted (guarded by insertValidator).